alacritty_terminal/src/term/mod.rs

```diff
@@ -2109,14 +2089,26 @@ impl<T: EventListener> Handler for Term<T> {
         self.cursor_style = style;
     }
 
+    #[inline]
+    fn set_title(&mut self, title: &str) {
+        if self.dynamic_title {
+            let title = title.trim().to_owned();
+
+            trace!("Setting window title to '{}'", title);
+
+            self.title = Some(title.clone());
+            self.event_proxy.send_event(Event::Title(title));
+        }
+    }
+
     #[inline]
     fn push_title(&mut self) {
-        trace!("Pushing '{}' onto title stack", self.title);
+        trace!("Pushing '{:?}' onto title stack", self.title);
```

Comment:
Could we just avoid this `None` thing and push default title? I don't really like `Option<String>` for title stack and for title. We can just set `self.title = self.default_title` and remove all the `Option<String>` thing.

Reply:
You're asking to make it impossible to change the title after it has been pushed and retrieved from the stack then?

We can't 'just' change it unless we want to stop supporting this usecase.

Reply:
I've likely commented in a wrong place, treat this comment as a general PR thing. I feel like we don't need things to be `Option`, that's what I'm trying to say.

Reply:
That's what I'm saying though. If you want the functionality to behave the way it does right now, we do need it to be an `Option`.

So the question is just if we want to have it the way it is right now and accept the ugly code, or if we want to have worse functionality for the user but cleaner code.

Don't get me wrong, I don't love this option either, I'm fine with `title: Option<String>`, but I think the `title_stack: Vec<Option<String>>` is pretty bad. I was kinda hoping for a comment related to this, but I'm not sure myself what I would prefer. But having a nice reloading functionality that works the way the user would expect definitely seems nice.

Reply:
The data types **must** match what we want the expected behavior to be, never the other way around. That being said, I'm not convinced that we can't have the best of both worlds still...

In the world without both `title` and `title_stack`, I feel like we could have a single `title_stack: Vec<String>` which when empty means "use default" much like how `None` means use default today. I see no value in pushing `None` values to a title stack, other than a minuscule amount of memory saved.

Reply:
Can't do that, we're still not allowed to pop back to the default title.